common/flatpak-progress.c

```diff
@@ -192,6 +192,7 @@ update_status_progress_and_estimating (FlatpakProgress *self)
      extra data, so ignore those */
   if (self->requested == 0)
     {
+      g_string_free (buf, TRUE);
```

Comment:
Alternatively, move the allocation to after the check.

Reply:
Yeah that seems cleaner